packages/fetchai/skills/carpark_client/dialogues.py

```diff
@@ -82,22 +72,28 @@ def __init__(self, **kwargs) -> None:
         Model.__init__(self, **kwargs)
         FipaDialogues.__init__(self, agent_address=self.context.agent_address)
 
-    def _create_dialogue(
-        self,
-        dialogue_label: DialogueLabel,
-        agent_address: Address,
-        role: BaseDialogue.Role,
+    @staticmethod
+    def role_from_first_message(message: Message) -> Dialogue.Role:
+        """
+        Infer the role of the agent from an incoming or outgoing first message
+
+        :param message: an incoming/outgoing first message
+        :return: the agent's role
+        """
+        return FipaDialogue.AgentRole.BUYER
+
+    def create_dialogue(
```

Comment:
Reimplementation as we need the derived class `Dialogue` which has been extended above...